Assets/LeapMotion/Scripts/Utils/Hands.cs

```diff
@@ -0,0 +1,169 @@
+﻿using UnityEngine;
+using System.Collections.Generic;
+using UnityEngine.SceneManagement;
+
+namespace Leap.Unity {
+
+  public static class Hands {
+
+    private static LeapServiceProvider s_provider;
+    private static GameObject s_leapRig;
+
+    static Hands() {
+      InitStatic();
+      SceneManager.activeSceneChanged += InitStaticOnNewScene;
```

Comment:
So I think that this can catch some of the cases, but it still doesn't catch the general case of swapping out the provider for a different one.  It also would not work if the provider is spawned 1 frame after the scene changes (like how can happen with blocks due to the networking).  

I think the only failsafe way would be some kind of getter that automatically checks to see if the current reference is null, and if so go searching for a new one.-